### PR TITLE
US138681 - Quizzing - Null question score validation fixes

### PR DESCRIPTION
**Pull request by maddiegabriel**
[US138681 - [Consistent Evaluation][Quizzing][UI] Bug fixes with setting null as score for QUESTION SCORE](https://rally1.rallydev.com/#/42960320374d/dashboard?detail=%2Fuserstory%2F633568672801)

This PR addresses the issues found in the question-level scorebox validation for quizzing.

It depends on this PR in `grade-result`: https://github.com/BrightspaceUILabs/grade-result/pull/93

In quizzing we don't allow a blank score for a question. The red error validation tooltip wasn't appearing when the user cleared the score box in some cases. Particular issues were found when going from a 0 to `undefined` (blank) score (might be related to this: https://github.com/BrightspaceUI/core/pull/2220/files). There were also some issues with the error tooltip sticking when switching between students/attempts which are resolved.

Outstanding issue: when the user clears a non-zero score, the scorebox is auto-populated with a zero. Ideally the scorebox would stay blank to avoid the error persisting if the user then wants to enter a zero.

This PR will fix the same thing in the RHP grade box for quizzing: https://github.com/BrightspaceHypermediaComponents/consistent-evaluation/pull/1000

GIF of current behaviour:

https://user-images.githubusercontent.com/26232541/167516413-7eb26afa-6d03-4be0-9d1a-73ef34729c96.mp4

GIF of switching between attempts:

https://user-images.githubusercontent.com/26232541/167516593-5fbbe2cb-6e6b-42e8-aad5-a5a887e6d196.mp4

### CODE REVIEW

File: components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js [hunk 1]
```diff
@@ -118,6 +118,7 @@ export class ConsistentEvaluationQuizzingAttemptResult extends SkeletonMixin(Loc
 			if (questionResponseEntity && questionResponseEntity.entity) {
 				this._savedTime = questionResponseEntity.entity.properties.savedTime;
 			}
+			this._scoreValidationError = undefined;
```
Comment (maddiegabriel) [inline components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js:121]: Set error tooltip back to undefined when we reload the LHP (this is what fixed the sticky issue between students/attempts)

File: components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js [hunk 2]
```diff
@@ -190,6 +191,7 @@ export class ConsistentEvaluationQuizzingAttemptResult extends SkeletonMixin(Loc
 						scoreNumerator=${this.score}
 						scoreDenominator=${this.outOf}
 						subtitleText=${this.localize('score')}
+						disallow-null
```
Comment (maddiegabriel) [inline components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js:194]: The `d2l-labs-d2l-grade-result-presentational` component should only render the error tooltip for undefined (blank) scores if we set this disallow-null property when rendering the component.

File: components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js [hunk 3]
```diff
@@ -186,6 +187,7 @@ export class ConsistentEvaluationQuizzingAttemptResult extends SkeletonMixin(Loc
 			<div class="d2l-consistent-eval-quiz-question-score">
 				<div class="d2l-consistent-eval-quiz-grade-result">
 					<d2l-labs-d2l-grade-result-presentational
+						disallow-null
```
Comment (maddiegabriel) [inline components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js:190]: Adding this property to `d2l-labs-d2l-grade-result-presentational` in this PR: https://github.com/BrightspaceUILabs/grade-result/pull/93

File: components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js [hunk 4]
```diff
@@ -222,8 +225,19 @@ export class ConsistentEvaluationQuizzingAttemptResult extends SkeletonMixin(Loc
 	_saveOnScoreChange(e) {
 		const newScore = e.detail.value;
 		this.score = newScore;
-		const isValidScore = typeof newScore === 'number' && newScore >= 0 && newScore <= 9999999999;
-		this._scoreValidationError = isValidScore ? undefined : this.localize('quizQuestionInvalidScore');
+		let isValidScore;
+
+		if (typeof newScore !== 'number') {
+			this._scoreValidationError = this.localize('quizQuestionNullScore');
```
Comment (maddiegabriel) [inline components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js:231]: This is the error message which will be displayed in the error toolip

**Review by Scy-D2L — APPROVED** [review]

**Review by JamieTran — APPROVED** [review]

**Pull request merged**